### PR TITLE
Development time tracker for bf and activity

### PR DESCRIPTION
**Pull request by AmanPrasad43**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [x] PR description and commit message should describe the changes done in this PR
- [x] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
	- Introduced development time tracking for activities and business flows, allowing users to track the time spent on development tasks within the application.
- **Bug Fixes**
	- Enhanced XML content length check to accommodate new development time tracking data.
- **Tests**
	- Added new test methods for verifying development time tracking functionality for business flows and activities.
- **Improvements**
	- Added event handling for unselecting items in the business flow tree, ensuring proper subscription management to avoid duplicates.

### CODE REVIEW

File: Ginger/GingerCoreCommon/Repository/BusinessFlowLib/BusinessFlow.cs [hunk 1]
```diff
@@ -61,6 +65,17 @@ public BusinessFlow(string sName)
             Activities.CurrentItem = a;
             CurrentActivity = a;
             AllowAutoSave = true;
+            _stopwatch = new Stopwatch();
```
Comment (Maheshkale447) [inline Ginger/GingerCoreCommon/Repository/BusinessFlowLib/BusinessFlow.cs:68]: Remove it and use it from start timer method

File: Ginger/GingerCoreCommon/Repository/BusinessFlowLib/BusinessFlow.cs [hunk 2]
```diff
@@ -104,6 +119,39 @@ public enum eSource
...
 
+        private TimeSpan mDevelopmentTime;
+        [IsSerializedForLocalRepository]
+        public TimeSpan DevelopmentTime
+        {
+            get
+            {
+                StopTimer();
+                return mDevelopmentTime;
+            }
```
Comment (Maheshkale447) [inline Ginger/GingerCoreCommon/Repository/BusinessFlowLib/BusinessFlow.cs:130]: add setter for this property and use it instead of `mDevelopmentTime`

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs [hunk 3]
```diff
@@ -122,6 +133,13 @@ void ITreeViewItem.SetTools(ITreeView TV)
                     TreeViewUtils.AddMenuItem(mContextMenu, "Automate", Automate, null, eImageType.Automate);
                 }
 
+                if (!isUnselectedSubscribed)
+                {
+                    TreeViewItem.Unselected += TreeViewItem_Unselected;
```
[the remark this answers is not in the document]
Comment (AmanPrasad43) [reply Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs]: handled

File: Ginger/GingerCoreCommon/Repository/BusinessFlowLib/Activity.cs [hunk 4]
```diff
@@ -127,13 +130,59 @@ public Activity()
...
+        private TimeSpan mDevelopmentTime;
+
+        [IsSerializedForLocalRepository]
+        public TimeSpan DevelopmentTime
+        {
+            get
+            {
+                StopTimer();
+                return mDevelopmentTime;
+            }
```
[the remark this answers is not in the document]
Comment (AmanPrasad43) [reply Ginger/GingerCoreCommon/Repository/BusinessFlowLib/Activity.cs:157]: done

File: Ginger/GingerCoreCommon/Repository/BusinessFlowLib/BusinessFlow.cs (hunk 1 above)
Comment (AmanPrasad43) [reply Ginger/GingerCoreCommon/Repository/BusinessFlowLib/BusinessFlow.cs]: handled

File: Ginger/GingerCoreCommon/Repository/BusinessFlowLib/BusinessFlow.cs (hunk 2 above)
Comment (AmanPrasad43) [reply Ginger/GingerCoreCommon/Repository/BusinessFlowLib/BusinessFlow.cs:128]: handled

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**